src/modules/xiv/rotationBuilder/__tests__/calc.spec.ts

```diff
@@ -0,0 +1,120 @@
+// import { LightningElement } from 'lwc';
+// import { getActionInfo } from 'xiv/actionRepository';
+// import { getJobNames } from 'xiv/actionRepository';
+// import { getJobActions } from 'xiv/actionRepository';
+// import { JobGuide } from 'xiv/actionData';
+// import HelloWorldApp  from '../rotationBuilder';
+// import { parseEffect } from '../../actionData/parseEffect';
+
+import { DefaultJobResources } from "xiv/defaultSnippets";
+import { getActionInfo } from "xiv/actionRepository";
+import { JobGuide } from "../../actionData/actionData";
+
+// Handled undefined actions
+// Undefined: return 0, otherwise return action's potency
+function getPotency(action, context): number {
+    if(!action) {
+        console.warn(`Undefined action info for ${context}`);
+        return 0;
+    }
+
+    return parseFloat(action.comboPotency || action.potency || 0);
+}
+
+// Test the potency of a specified snippet for a given job
+function testSnippetPotency(testName, snippetName, jobKey, jobName, version, expectedPotency): void {
+    if(!JobGuide[jobName]) {
+        console.warn(`Undefined job info for ${jobName}`);
+        return;
+    }
+
+    test(testName, () => {
+        const jobSnippet = DefaultJobResources[jobKey].snippets.find((s) => s.name === snippetName);
+        const actionList = jobSnippet.versions[version].actions.map((snippetAction) => {
+            const actionInfo = getActionInfo(jobName, snippetAction.action);
+            return {actionInfo, actionName: snippetAction.action};
+        });
+
+        //console.log("Actions: ", actionList);
+
+        const sumPotency = actionList.reduce((accum, {actionInfo, actionName}) => {
+            return accum + getPotency(actionInfo, `Action: ${actionName}`);
```

Comment:
While this does calculate the potency sum, it doesn't test the calculate function that will be used by the optimizer.